app/views/declaration/declaration_holder_add.scala.html

```diff
@@ -0,0 +1,87 @@
+@*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import controllers.declaration.routes._
+@import forms.declaration.DeclarationHolder
+@import services.HolderOfAuthorisationCode
+@import services.view.AutoCompleteItem
+@import models.requests.JourneyRequest
+@import controllers.navigation.Navigator
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import views.{BackButton, Title}
+@import views.html.components.gds._
+
+@this(
+        govukLayout: gdsMainTemplate,
+        govukButton: GovukButton,
+        govukTable : GovukTable,
+        errorSummary: errorSummary,
+        sectionHeader: sectionHeader,
+        pageTitle: pageTitle,
+        exportsInputText: exportsInputText,
+        tariffDetails: tariffDetails,
+        addButton: addButton,
+        removeButton: removeButton,
+        saveButtons: saveButtons,
+        formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+)
+
+@(mode: Mode, form: Form[DeclarationHolder], holders: Seq[DeclarationHolder])(implicit request: JourneyRequest[_], messages: Messages)
+
+@tariffBody = { @Html(messages("declaration.type.holderOfAuthorisationSummaryText", components.details_content_link())) }
+
+@backLink = @{
+    if(holders.nonEmpty)
+        DeclarationHolderController.displayPage(mode)
+    else
+        Navigator.backLink(DeclarationHolder, mode)
```

Comment:
Should this logic be in the Navigator? Similar like clearance requests helper methods?

Reply:
Possibly.  But I'd have to create a brand new `DeclarationPage` object for this. `DeclarationHolder` is the Navigator key for navigating from the summary page.  The other "helper" pages - add, remove, change - all always go back to the summary page, except this one instance (no existing holders)

Reply:
Created new `DeclarationPage` object and moved logic to Navigator.  This allowed me to change the view to stop passing in the existing holders collection just for the purpose of back-link creation